ci-operator/step-registry/openshift/image-based/infra/sos/openshift-image-based-infra-sos-commands.sh

```diff
@@ -0,0 +1,33 @@
+#!/bin/bash
+set -euo pipefail
+
+trap 'CHILDREN=$(jobs -p); if test -n "${CHILDREN}"; then kill ${CHILDREN} && wait; fi' TERM
+
+IP_ADDRESS="$(cat "${SHARED_DIR}/public_address")"
+HOST_USER="$(cat "${SHARED_DIR}/ssh_user")"
+INSTANCE_PREFIX="${HOST_USER}@${IP_ADDRESS}"
+
+echo "Using Host $IP_ADDRESS"
+
+SSHOPTS=(-o 'ConnectTimeout=5'
+  -o 'StrictHostKeyChecking=no'
+  -o 'UserKnownHostsFile=/dev/null'
+  -o 'ServerAliveInterval=90'
+  -o LogLevel=ERROR
+  -i "${CLUSTER_PROFILE_DIR}/ssh-privatekey")
+
+ssh "${SSHOPTS[@]}" "${INSTANCE_PREFIX}" <<'EOF'
+  if ! hash sos ; then
+    sudo touch /tmp/sosreport-command-does-not-exist
+    exit 0
+  fi
+
+  plugin_list="container,network"
+
+  if sudo sos report --batch --all-logs --tmp-dir /tmp -p ${plugin_list} -o logs ; then
+    sudo chmod +r /tmp/sosreport-*
+  else
+    sudo touch /tmp/sosreport-command-failed
+  fi
+EOF
+scp "${SSHOPTS[@]}" "${INSTANCE_PREFIX}":/tmp/sosreport-* "${ARTIFACT_DIR}" || true
```

Comment:
In the context of the command you've provided, || true is used to ensure that the command preceding it (scp in this case) does not cause the script or command line to exit with a non-zero status, even if it fails. 

I think this is not desired, but im not sure.

Reply:
That is desired I believe because I think there are cases we might not get an SOS report and in that case we don't want the scp to fail if the file doesn't exist

Reply:
I understand the use case now, but if we cannot connect to the instance then it wouldnt fail as well. Im wondering if theres a way for us to do this smarter but for now thats fine :)